ELMOD1.py

```diff
@@ -0,0 +1,317 @@
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from Bio.SeqRecord import SeqRecord
+from typing import List, Dict, Union, Tuple
+from abc import ABC, abstractmethod
+import os
+
+
+class FastQFilter:
+    def __init__(self, input_path: str, output_filename: str, gc_bounds: Union[int, Tuple[int, int]] = (0, 100),
+                 length_bounds: Union[int, Tuple[int, int]] = (0, 2**32), quality_threshold: int = 0):
+        """
+        Initialize FastQFilter instance.
+
+        Parameters:
+        - input_path (str): Path to the input FastQ file.
+        - output_filename (str): Name for the output FastQ file.
+        - gc_bounds (Union[int, Tuple[int, int]]): GC content bounds for filtering.
+        - length_bounds (Union[int, Tuple[int, int]]): Length bounds for filtering.
+        - quality_threshold (int): Quality threshold for filtering.
+        """
+        self.input_path = input_path
+        self.output_filename = output_filename
+        self.gc_bounds = gc_bounds
+        self.length_bounds = length_bounds
+        self.quality_threshold = quality_threshold
+
+    def filter_fastq(self) -> None:
+        """
+        Read, filter, and write FastQ records based on specified criteria.
+        """
+        records = self.read_fastq()
+        filtered_records = self.apply_filters(records)
+        self.write_fastq(filtered_records)
+
+    def read_fastq(self) -> List[SeqRecord]:
+        """
+        Read FastQ file and return a list of SeqRecord objects.
+        """
+        records = list(SeqIO.parse(self.input_path, "fastq"))
+        return records
```

Comment:
Это уже из разряда "что можно сделать, чтобы вообще конфетка была". Можно добавить блок исключений при чтении fastq файла для более корректной обработки возможных ошибок ввода-вывода

```suggestion
    def read_fastq(self) -> List[SeqRecord]:
    """
    Read FastQ file and return a list of SeqRecord objects.
    """
    try:
        records = list(SeqIO.parse(self.input_path, "fastq"))
        return records
    except IOError as e:
        print(f"Error reading FastQ file: {e}")
        return []
```